python/lsst/verify/extract_metricvalues.py

```diff
@@ -0,0 +1,243 @@
+                return False
+        return True
+
+    def value_formatter(value):
+        return f"{value}"
+
+    match kind:
+        case "value":
+            result = load_value(butler, verbose=verbose)
+        case "timing":
+            result = load_timing(butler, verbose=verbose)
+
+            def value_formatter(value):
+                return f"{value.datum.label}: {value.quantity:.4}"
+        case "memory":
+            result = load_memory(butler, verbose=verbose)
+
+            def value_formatter(value):
+                return f"{value.datum.label}: {value.quantity.to(u.Mbyte):.5}"
+        case _:
+            raise RuntimeError(f"Cannot handle kind={kind}")
+
+    for (data_id, metric), value in sorted(result.items()):
+        if data_id_restriction is not None and not match_data_id(data_id):
+            continue
+
+        print(f"{_data_id_label(data_id, data_id_keys)};", value_formatter(value))
+
+
+def print_diff_metrics(butler1, butler2, data_id_keys=None, verbose=None):
+    """Load metric values from two repos and print their differences.
+
+    This only supports differencing metrics that aren't time or memory-related.
+
+    Parameters
+    ----------
+    butler1, butler2 : `lsst.daf.butler.Butler`
+        Butlers to load values to difference from.
+    data_id_keys : `tuple` [`str`], optional
+        List of Butler dataId keys to restrict the printed output to;
+        for example: ``("detector", "visit")``.
+    verbose : `bool`, optional
+        Print extra information when loading values, and about failures.
+    """
+    result1 = load_value(butler1)
+    result2 = load_value(butler2)
+
+    same = 0
+    failed = 0
+    for key in result1:
+        data_id, metric = key
+        try:
+            value1 = result1[key]
+            value2 = result2[key]
+        except KeyError:
+            print(f"Result 2 does not contain metric '{metric}'' for dataId '{data_id}'")
+            failed += 1
+            continue
+
+        delta = value1.quantity - value2.quantity
+        if delta != 0 or verbose:
+            print(f"{_data_id_label(data_id, data_id_keys)};", value1.datum.label, delta)
+        if delta == 0:
+            same += 1
+
+    if same > 0:
+        print(f"Number of metrics that are the same in both runs, out of {len(result2)}: {same}")
+
+    if failed != 0:
+        keys1 = sorted(list(result1.keys()))
+        keys2 = sorted(list(result2.keys()))
+        print()
+        print(f"butler1 metrics found: {len(result1)}")
+        print(f"butler2 metrics found: {len(result2)}")
+        print(f"metrics in butler1 that were not found in butler2: {failed}")
+        print("Check that the butler schemas are comparable, if you expected one-to-one comparability.")
+        if verbose:
+            print("Full DataCoordinates for the first key of each result, to compare schemas:")
+            print(keys1[0][0].full)
+            print(keys2[0][0].full)
```

Comment:
butler registry schemas determine whether you can actually compare metrics in different repos. I discovered this when trying to compare different `ap_verify` runs, one before the `visit_system` registry change, and one after. It reported that nothing was found, because the dataIds weren't comparable. Instead of trying to hack together a way to make them comparable (it's not at all easy, per slack discussion), I just added this message.

See more here: https://lsstc.slack.com/archives/C01FBUGM2CV/p1654727269705119

Reply:
I'm asking about the opposite, though -- if a metric value dataset is found in one repository but not another, why would you conclude that it's a schema change instead of the dataset really not being there?

Reply:
I'm not concluding that, it's just something the user has to check: it really confused me when nothing was found. I've reworded the message a bit.

Not having any more usecases than ap_verify, I don't really know what the corner cases here are supposed to be. I'd rather not worry about those things until this gets some broader use.